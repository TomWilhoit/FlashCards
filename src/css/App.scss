.App {
  text-align: center;
}

.review-header {
  display: flex;
  justify-content: center;
}

.question {
  display: flex;
  justify-content: center;
}
