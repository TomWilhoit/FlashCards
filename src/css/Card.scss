.false-answer-btn1, 
.false-answer-btn2,
.correct-answer-btn{
  height: 50px;
  margin: 20px;
  font-size: 25px;
  border-radius: 10px;
}