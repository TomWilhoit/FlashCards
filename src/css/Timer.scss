.timer-clock {
  font-size: 70px;
  margin: 35px;
  color: crimson;
}