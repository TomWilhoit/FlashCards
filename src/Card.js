import React, { Component } from 'react';
import './css/App.css';

class Card extends Component {
  render() {
    return (
      <div className="Card">
        <h2>Card Test</h2>
      </div>
    );
  }
}





export default Card;