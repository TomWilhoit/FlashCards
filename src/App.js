import React, { Component } from 'react';
import CardCountainer from './CardContainer';
import Timer from './Timer';
import Banner from './Banner';
import Counter from './Counter';
import Card from './Card';
import FlashCardDataSet from './FlashCardDataSet';
import './css/App.css';

class App extends Component {
  constructor() {
    super();


    this.state = {
      questions: FlashCardDataSet,
      shouldRestart: false,
      questionIndex: 0,
      savedArray: [],
    }
  }
 
  shouldRestart = () => {
    if (this.state.shouldRestart===false) {
    this.setState({
      shouldRestart: true
    })
  } else{
    this.setState({
      shouldRestart: false
      })
    }
  }

  incrementQuestionIndex = () => {
    const newIndex = this.state.questionIndex + 1;
    this.setState({
      questionIndex: newIndex
    })
  }

  saveToStorage = () => {
    let [...savedQuestions] = this.state.savedArray
    savedQuestions.push(this.state.questions[this.state.questionIndex])
    localStorage.setItem('savedQuestions',JSON.stringify(savedQuestions))
    this.setState({
      savedArray: savedQuestions
    })

      

    }
  
  
  
  render() {
    return (
      <div className="App">
        <Banner/>
        <CardCountainer
        questionObj= {this.state.questions[this.state.questionIndex]}
        incrementQuestionIndex = {this.incrementQuestionIndex}
        saveToStorage = {this.saveToStorage}/>
        <Timer shouldRestart={this.shouldRestart}/>
        <Counter
        questionIndex = {this.state.questionIndex}/>

      </div>
    ) };
  }

export default App;
