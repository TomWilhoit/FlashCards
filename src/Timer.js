import React, { Component } from 'react';
import './css/App.css';
import App from './App';


export default class Timer extends Component {
  constructor(props) {
    super(props)
    
    this.state = {
      seconds: 900
    };
  }
  
  componentWillMount() {
    this.startTimer()
  }

  restartTimer = () => {
    this.setState({
      seconds: 900
    })
    this.props.shouldRestart();
  }
  
  startTimer = () => {
    this.timer = setInterval(this.eachTick, 1000);
  }
  

  eachTick = () => {
    if (this.state.seconds > 0)
    this.setState({
      seconds: (this.state.seconds - 1)
    });

    if (this.state.seconds === 0){
      console.log('time up');
      clearInterval(this.timer);
      this.props.shouldRestart();

    }

  }

  render() {
    return (
      <div className='timer-container'>
        <h2 className='timer-clock'>{this.state.seconds}</h2>
        <button className='start-btn' onClick={this.startTimer}>Start Timer</button>
        <button className='restart-btn' onClick={this.restartTimer}>Restart Button</button>
      </div>
    );
  }
}
  




