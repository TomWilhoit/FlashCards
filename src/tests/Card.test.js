import React from 'react';
import Timer from '../Timer';
import { shallow } from 'enzyme';
import Card from '../Card';
import App from '../App';

let wrapper;

describe('Card', () => {
 
  beforeEach(() => {
  wrapper = shallow(
    <Card/>
    );
  });
  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({
      correctClicked: false,
      questionOrder: 1
    });
  });
});
