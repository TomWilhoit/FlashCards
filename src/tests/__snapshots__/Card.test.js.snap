// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot when all data is passed correctly 1`] = `
<div
  className="Card"
>
  <h2 />
  <button
    className="correct-answer-btn"
    onClick={[Function]}
  />
  <button
    className="false-answer-btn1"
    onClick={[Function]}
  />
  <button
    className="false-answer-btn2"
    onClick={[Function]}
  />
</div>
`;
